core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -655,6 +667,29 @@ public boolean canDeleteIndexContents(Index index, IndexSettings indexSettings,
     }
 
     /**
+     * This method returns true if the current node is allowed to delete the given index's
+     * metadata folders only, but not the shard data. This is the case if the index is on a
+     * shared file system, and it is deleted without having been closed before, in which case
+     * the shared shard data should not be deleted because other nodes would have data structures
+     * depending on that shared data.
+     * @param index {@code Index} to check whether deletion is allowed
+     * @param indexSettings {@code IndexSettings} for the given index
+     * @param closed true if the index has been deleted closed already, but not yet deleted
+     * @return true if the index can be deleted on this node
+     */
+    public boolean canDeleteIndexMetadataOnly(Index index, IndexSettings indexSettings, boolean closed) {
```

Comment:
this is currently a problem as well, right? what deletes the folder now when we use, for example, dedicated master nodes? note that it's enough to delete the index state/metadata files to prevent dangling indices.

As said - if this is an existing shadow indexes problem, we need to document it via an issue/test - not solve it.